internal/events/loop.go

```diff
@@ -88,82 +83,84 @@ func (el *EventLoop) handleEvent(ctx context.Context, event interface{}) error {
 		return err
 	}
 
-	el.processChangesAndStatusUpdates(ctx, changes, updates)
+	changed, conf, statuses := el.processor.Process()
+	if !changed {
+		return nil
+	}
+
+	err = el.updateNginx(ctx, conf)
+	if err != nil {
+		el.logger.Error(err, "Failed to update NGINX configuration")
+	}
+
+	// FIXME(pleshakov) Update resource statuses instead of printing to stdout
+	for name, s := range statuses.ListenerStatuses {
+		fmt.Printf("Listener %q, Statuses: %v\n", name, s)
+	}
+	for nsname, s := range statuses.HTTPRouteStatuses {
+		fmt.Printf("HTTPRoute %q, Statuses: %v\n", nsname.String(), s)
+	}
+
 	return nil
 }
 
-func (el *EventLoop) propagateUpsert(e *UpsertEvent) ([]state.Change, []state.StatusUpdate, error) {
+func (el *EventLoop) updateNginx(ctx context.Context, conf newstate.Configuration) error {
+	cfg, warnings := el.generator.Generate(conf)
+
+	// For now, we keel all http servers in one config
+	// We might rethink that. For example, we can write each server to its file
+	// or group servers in some way.
+	err := el.nginxFileMgr.WriteHTTPServersConfig("http-servers", cfg)
+	if err != nil {
+		return err
+	}
+
+	for obj, objWarnings := range warnings {
+		for _, w := range objWarnings {
+			// FIXME(pleshakov): report warnings via Object status
+			el.logger.Info("got warning while generating config",
+				"kind", obj.GetObjectKind().GroupVersionKind().Kind,
+				"namespace", obj.GetNamespace(),
+				"name", obj.GetName(),
+				"warning", w)
+		}
+	}
+
+	return el.nginxRuntimeMgr.Reload(ctx)
+}
+
+func (el *EventLoop) propagateUpsert(e *UpsertEvent) error {
 	switch r := e.Resource.(type) {
+	case *v1alpha2.Gateway:
+		el.processor.CaptureUpsertChange(r)
+		return nil
 	case *v1alpha2.HTTPRoute:
-		changes, statusUpdates := el.conf.UpsertHTTPRoute(r)
-		return changes, statusUpdates, nil
+		el.processor.CaptureUpsertChange(r)
+		return nil
 	case *apiv1.Service:
 		el.serviceStore.Upsert(r)
 		// FIXME(pleshakov): make sure the affected hosts are updated
-		return nil, nil, nil
+		return nil
 	}
 
 	// FIXME(pleshakov): panic because it is a coding error
```

Comment:
Related to my previous comment, why is this a FIXME? What's stopping us from changing this to a panic now?

Reply:
fixed fixmes. I was holding it off because it wasn't related to this PR and the parent PR. But it is a not big change , 
 but at the same time it will make things less confusing, so added it.